### ISSUE TITLE
data-collector-3.9.0 contains references to both AndroidX and old support library

### ISSUE THREAD

**Issue by plastiv** [braintree/braintree_android#302]
### Issue description

> WARNING: [Processor] Library '.gradle/caches/modules-2/files-2.1/com.paypal.android.sdk/data-collector/3.9.0/fb5f4d069ba36bb7bb19e4cc9ba7c299bd896902/data-collector-3.9.0.aar' contains references to both AndroidX and old support library. This seems like the library is partially migrated. Jetifier will try to rewrite the library anyway.
>   Example of androidX reference: 'androidx/annotation/NonNull'
>   Example of support library reference: 'android/support/annotation/NonNull'

```
com.paypal.android.sdk:data-collector:3.9.0
-- com.paypal.android.sdk:paypal-one-touch:3.9.0
---- com.braintreepayments.api:braintree:3.9.0
```

**Comment by sshropshire** [comment]
Hi @plastiv thank you for using the Braintree SDK for Android. I'm attempting to reproduce this issue on my workstation. Can you provide either Android Studio steps or the `gradlew` command you used that caused the warning to appear?

**Comment by plastiv** [comment]
Hi @sshropshire thank you for taking a look on it.

I've noticed the warning after AGP 4.0 update. I think warning is connected with jettifier version bundled into it. I don't know the way on how to run jettifier on certain dependency so I just created new project with AS4 and then added `implementation 'com.braintreepayments.api:braintree:3.9.0'` there. 

You can find a sample here: https://github.com/plastiv/braintree-jetifier-issue/runs/755367178?check_suite_focus=true#step:4:30

**Comment by sshropshire** [comment]
@plastiv I'm having trouble reproducing looks like the CI is calling `./gradlew assembleDebug`. Does this occur on your dev machine as well?

**Comment by plastiv** [comment]
Yes I can reproduce it locally. 

Warning is shown when jetifier runs over dependency. Processed artifact is later cached and output is reused which is likely a reason you don't see warning locally but on ephemeral CI only. To avoid dealing with cached artifacts you can run jetifier as standalone tool: https://developer.android.com/studio/command-line/jetifier 

1. download and extract jetifier https://dl.google.com/dl/android/studio/jetifier-zips/1.0.0-beta09/jetifier-standalone.zip
2. download .aar https://repo1.maven.org/maven2/com/paypal/android/sdk/data-collector/3.9.0/data-collector-3.9.0.aar
3. run `./jetifier-standalone -i data-collector-3.9.0.aar -o out -l verbose`

Which gives similar output:

```
VERBOSE: [Config] Using the default config '/default.generated.config'
INFO: [Archive] Extracting: /Downloads/jetifier-standalone/bin/data-collector-3.9.0.aar
VERBOSE: [Archive] Extracting archive: AndroidManifest.xml
VERBOSE: [Archive] Extracting archive: R.txt
INFO: [Archive] Extracting nested: classes.jar
VERBOSE: [Archive] Extracting archive: com/paypal/android/sdk/BuildConfig.class
VERBOSE: [Archive] Extracting archive: com/paypal/android/sdk/data/collector/InstallationIdentifier.class
VERBOSE: [Archive] Extracting archive: com/paypal/android/sdk/data/collector/PayPalDataCollector.class
VERBOSE: [Archive] Extracting archive: com/paypal/android/sdk/data/collector/PayPalDataCollectorRequest.class
INFO: [Archive] Extracting nested: libs/paypal-data-collector-4.1.2.jar
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/R.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesSettings$Builder.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c$e.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/InvalidInputException.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesSource.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/d$1.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/Environment.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesSettings$1.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c$c.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c$a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesResult.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/i.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworkingFactory.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworking.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/k.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/o.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/m.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/j.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/MagnesNetworkingFactoryImpl.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/h.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/l.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/n.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/b.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/d.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/d$a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/f.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/c.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/g.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/PayPalCertificate.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/network/e.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c$d.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/a/b.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/a/a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/a/c.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c$b.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/b.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/d.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/d$a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesSettings.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/MagnesSDK.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/c.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/d$b.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/b/a$a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/b/a.class
VERBOSE: [Archive] Extracting archive: lib/android/paypal/com/magnessdk/e.class
INFO: [Processor] Started new transformation
INFO: [Processor] - Input file: data-collector-3.9.0.aar
VERBOSE: [Processor] [Applied: XmlResourcesTransformer] AndroidManifest.xml
VERBOSE: [Processor] [Skipped] R.txt
VERBOSE: [Processor] [Applied: ByteCodeTransformer] com/paypal/android/sdk/BuildConfig.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] com/paypal/android/sdk/data/collector/InstallationIdentifier.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] com/paypal/android/sdk/data/collector/PayPalDataCollector.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] com/paypal/android/sdk/data/collector/PayPalDataCollectorRequest.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/R.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesSettings$Builder.class
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/Size -> androidx/annotation/Size
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c$e.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/InvalidInputException.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesSource.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/d$1.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/Environment.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesSettings$1.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c$c.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c$a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesResult.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/i.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworkingFactory.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworking.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/k.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/o.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/m.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/j.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/MagnesNetworkingFactoryImpl.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/h.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/l.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/n.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/b.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/d.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/d$a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/f.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/c.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/g.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/PayPalCertificate.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/network/e.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c$d.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/a/b.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/a/a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/a/c.class
INFO: [TypeRewriter] Map: android/support/annotation/Nullable -> androidx/annotation/Nullable
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c$b.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/b.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/d.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/d$a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesSettings.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/MagnesSDK.class
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/Nullable -> androidx/annotation/Nullable
INFO: [TypeRewriter] Map: android/support/annotation/Size -> androidx/annotation/Size
INFO: [TypeRewriter] Map: android/support/annotation/Nullable -> androidx/annotation/Nullable
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/NonNull -> androidx/annotation/NonNull
INFO: [TypeRewriter] Map: android/support/annotation/Nullable -> androidx/annotation/Nullable
INFO: [TypeRewriter] Map: android/support/annotation/Size -> androidx/annotation/Size
INFO: [TypeRewriter] Map: android/support/annotation/Nullable -> androidx/annotation/Nullable
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/c.class
INFO: [TypeRewriter] Map: android/support/annotation/RequiresApi -> androidx/annotation/RequiresApi
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/d$b.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/b/a$a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/b/a.class
VERBOSE: [Processor] [Applied: ByteCodeTransformer] lib/android/paypal/com/magnessdk/e.class
INFO: [Archive] Deleting old output file
INFO: [Archive] Writing archive: file://Downloads/jetifier-standalone/bin/out
VERBOSE: [Archive] Writing file: AndroidManifest.xml
VERBOSE: [Archive] Writing file: R.txt
VERBOSE: [Archive] Writing file: classes.jar
VERBOSE: [Archive] Writing file: com/paypal/android/sdk/BuildConfig.class
VERBOSE: [Archive] Writing file: com/paypal/android/sdk/data/collector/InstallationIdentifier.class
VERBOSE: [Archive] Writing file: com/paypal/android/sdk/data/collector/PayPalDataCollector.class
VERBOSE: [Archive] Writing file: com/paypal/android/sdk/data/collector/PayPalDataCollectorRequest.class
VERBOSE: [Archive] Writing file: libs/paypal-data-collector-4.1.2.jar
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/R.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesSettings$Builder.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c$e.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/InvalidInputException.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesSource.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/d$1.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/Environment.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesSettings$1.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c$c.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c$a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesResult.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/i.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworkingFactory.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/httpclient/MagnesNetworking.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/k.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/o.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/m.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/j.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/MagnesNetworkingFactoryImpl.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/h.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/l.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/n.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/b.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/d.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/d$a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/f.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/c.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/g.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/PayPalCertificate.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/network/e.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c$d.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/a/b.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/a/a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/a/c.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c$b.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/b.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/d.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/d$a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesSettings.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/MagnesSDK.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/c.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/d$b.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/b/a$a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/b/a.class
VERBOSE: [Archive] Writing file: lib/android/paypal/com/magnessdk/e.class
```

As you can see `lib/android/paypal/com/magnessdk/MagnesSettings.class` uses support lib 

```
package lib.android.paypal.com.magnessdk;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Size;
import lib.android.paypal.com.magnessdk.network.MagnesNetworkingFactoryImpl;

public class MagnesSettings {
```

while the rest of `data-collector` code declares to be built with androidx https://github.com/braintree/braintree_android/blob/master/PayPalDataCollector/src/main/java/com/paypal/android/sdk/data/collector/PayPalDataCollector.java#L6 and so is highlighted by jetifier warning.

You need to rebuild Magnes 4.1.2 with androidX. 

Hope that helps

**Comment by ChristianBecker** [comment]
Hey @sshropshire , I'm running in the same issue with the latest Braintree Android SDK 3.14.1. The warning is
```
[Processor] Library '/Users/becker/.gradle/caches/modules-2/files-2.1/com.paypal.android.sdk/data-collector/3.14.1/2a9890d47e7fec2e001a5771e69a689294137dcc/data-collector-3.14.1.aar' contains references to both AndroidX and old support library. This seems like the library is partially migrated. Jetifier will try to rewrite the library anyway.
```

While your statement about Jetifier is true, the warning obviously states that the `data-collector-3.14.1.aar` library contains references to both legacy Android Support and new AndroidX libraries, which is kind of a weird state.

Do you plan to migrate the last part over to AndroidX anytime soon? The Braintree Android SDK is one of the last libraries that forces me to enable Jetifier at all.

**Comment by sshropshire** [comment]
@ChristianBecker @plastiv thank you for the feedback. We're looking to resolve this in the `data-collector` module.

**Comment by ersen-lw** [comment]
Hello, is there any update on this? We want to remove the Jetifier usage from the project to improve build times and this is the last dependency that requires it. Thank you.

**Comment by sarahkoop** [comment]
👋 We've released version `4.5.0` of the SDK that resolves this issue in the `data-collector` module
